### ISSUE TITLE
Ssl error when auto uploding a file

### ISSUE THREAD

**Issue by buehlefs** [nextcloud/android#5765]
### Steps to reproduce

Auto upload for images and contact backups fails with an ssl error. Also fails when retrying manually.

### Expected behaviour
- The app should be able to correctly establish a ssl connection to the nextcloud server

### Actual behaviour
- The connection cannot be established

### Can you reproduce this problem on https://try.nextcloud.com?

Could not reproduce the issue on the demo instance. But I still believe it is not a server configuration issue. The desktop synchronization still works. And the activities view in the app still correctly updates. Only dav operations seem to be affected as the list of files in a folder also cannot be refreshed.

### Environment data
Android version: 8.0.0

Device model: SHIFT5me

Stock or customized system: stock

Nextcloud app version: 3.11.0

Nextcloud server version: 18.0.3

#### Additional information

The nextcloud server uses a letsencrypt certificate. (PKCS #1 SHA-256 mit RSA-Verschlüsselung)

The server update from 18.0.2 happened 2020-03-24.

The client update from 3.10.? happened 2020-03-26.

The last succesful autoupload happened 2020-03-24 (according to the timestamp in the filename from the contacts backup).

The first failing autoupload happened 2020-03-25.

This suggests that there is a possibility of a server issue too.

#### Web server error log

There is no relevant log information at the server. The connection fails client side before it can be established.

#### Nextcloud log (data/nextcloud.log)

<details>
<summary>
Client log (click to expand)
</summary>

```
03-29 14:00:43.160   435   478 I libPerfService: perfUserGetCapability - value:26480
03-29 14:00:44.174   435   478 I libPerfService: perfUserGetCapability - value:26480
03-29 14:00:45.188   435   478 I libPerfService: perfUserGetCapability - value:26480
03-29 14:00:45.476   435   648 I vendor.mediatek.hardware.power@1.1-impl: notifyAppState pack:com.nextcloud.client, act:com.owncloud.android.ui.activity.UploadListActivity, pid:26480, state:1
03-29 14:00:45.621 26480  1051 I ServerNameIndicator: SSLSocket implementation: com.google.android.gms.org.conscrypt.Java8FileDescriptorSocket
03-29 14:00:45.621 26480  1051 I ServerNameIndicator: SNI done, hostname: ***.feste-ip.net
03-29 14:00:45.632 26480  1050 I ServerNameIndicator: SNI done, hostname: ***.feste-ip.net
03-29 14:00:45.648   439  1287 I BufferQueueProducer: [com.nextcloud.client/com.owncloud.android.ui.activity.UploadListActivity#0](this:0x783bd6a800,id:1345,api:1,p:26480,c:439) connect(P): api=1 producer=(26480:com.nextcloud.client) producerControlledByApp=true
03-29 14:00:46.208   435   478 I libPerfService: perfUserGetCapability - value:26480
03-29 14:00:47.236   435   478 I libPerfService: perfUserGetCapability - value:26480
03-29 14:00:48.257   435   478 I libPerfService: perfUserGetCapability - value:26480
03-29 14:00:48.817 26480  1051 V NativeCrypto: SSL handshake aborted: ssl=0x7a9a8b4f88: I/O error during system call, Connection reset by peer
03-29 14:00:48.819 26480  1051 E GetUserInfoRemoteOperation: Exception while getting OC user information
03-29 14:00:48.819 26480  1051 E GetUserInfoRemoteOperation: javax.net.ssl.SSLHandshakeException: SSL handshake aborted: ssl=0x7a9a8b4f88: I/O error during system call, Connection reset by peer
03-29 14:00:48.819 26480  1051 E GetUserInfoRemoteOperation: 	at com.google.android.gms.org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)
03-29 14:00:48.819 26480  1051 E GetUserInfoRemoteOperation: 	at com.google.android.gms.org.conscrypt.NativeSsl.doHandshake(:com.google.android.gms@19530019@19.5.30 (040400-275531062):6)
03-29 14:00:48.819 26480  1051 E GetUserInfoRemoteOperation: 	at com.google.android.gms.org.conscrypt.ConscryptFileDescriptorSocket.startHandshake(:com.google.android.gms@19530019@19.5.30 (040400-275531062):13)
03-29 14:00:48.819 26480  1051 E GetUserInfoRemoteOperation: 	at com.owncloud.android.lib.common.network.AdvancedSslSocketFactory.verifyPeerIdentity(AdvancedSslSocketFactory.java:264)
03-29 14:00:48.819 26480  1051 E GetUserInfoRemoteOperation: 	at com.owncloud.android.lib.common.network.AdvancedSslSocketFactory.createSocket(AdvancedSslSocketFactory.java:187)
03-29 14:00:48.819 26480  1051 E GetUserInfoRemoteOperation: 	at org.apache.commons.httpclient.HttpConnection.open(HttpConnection.java:707)
03-29 14:00:48.819 26480  1051 E GetUserInfoRemoteOperation: 	at org.apache.commons.httpclient.MultiThreadedHttpConnectionManager$HttpConnectionAdapter.open(MultiThreadedHttpConnectionManager.java:1361)
03-29 14:00:48.819 26480  1051 E GetUserInfoRemoteOperation: 	at org.apache.commons.httpclient.HttpMethodDirector.executeWithRetry(HttpMethodDirector.java:387)
03-29 14:00:48.819 26480  1051 E GetUserInfoRemoteOperation: 	at org.apache.commons.httpclient.HttpMethodDirector.executeMethod(HttpMethodDirector.java:171)
03-29 14:00:48.819 26480  1051 E GetUserInfoRemoteOperation: 	at org.apache.commons.httpclient.HttpClient.executeMethod(HttpClient.java:397)
03-29 14:00:48.819 26480  1051 E GetUserInfoRemoteOperation: 	at org.apache.commons.httpclient.HttpClient.executeMethod(HttpClient.java:323)
03-29 14:00:48.819 26480  1051 E GetUserInfoRemoteOperation: 	at com.owncloud.android.lib.common.OwnCloudClient.executeMethod(OwnCloudClient.java:219)
03-29 14:00:48.819 26480  1051 E GetUserInfoRemoteOperation: 	at com.owncloud.android.lib.resources.users.GetUserInfoRemoteOperation.run(GetUserInfoRemoteOperation.java:93)
03-29 14:00:48.819 26480  1051 E GetUserInfoRemoteOperation: 	at com.owncloud.android.lib.common.operations.RemoteOperation.execute(RemoteOperation.java:136)
03-29 14:00:48.819 26480  1051 E GetUserInfoRemoteOperation: 	at com.owncloud.android.ui.activity.DrawerActivity$6.run(DrawerActivity.java:1030)
03-29 14:00:48.819 26480  1051 E GetUserInfoRemoteOperation: 	at java.lang.Thread.run(Thread.java:764)
03-29 14:00:48.829 26480  1050 V NativeCrypto: SSL handshake aborted: ssl=0x7aa4380f88: I/O error during system call, Connection reset by peer
03-29 14:00:49.280   435   478 I libPerfService: perfUserGetCapability - value:26480
03-29 14:00:50.307   435   478 I libPerfService: perfUserGetCapability - value:26480
03-29 14:00:50.934 26480 26480 D UploadListAdapter: loadUploadItemsFromDb
03-29 14:00:50.978 26480 26480 D FileUploader: Starting command with id 73
03-29 14:00:50.985 26480 26480 V UploadsStorageManager: Updating /storage/emulated/0/DCIM/**/*.jpg with status=UPLOAD_IN_PROGRESS
03-29 14:00:51.010 26480 26480 D UploadsStorageManager: updateUpload returns with: 1 for file: /storage/emulated/0/DCIM/**/*.jpg
03-29 14:00:51.010 26480 26480 D UploadsStorageManager: notifyObserversNow
03-29 14:00:51.038 26480 28560 V UploadsStorageManager: Updating /storage/emulated/0/DCIM/**/*.jpg with status:UPLOAD_IN_PROGRESS and result:UNKNOWN (old:/storage/emulated/0/DCIM/**/*.jpg status:UPLOAD_IN_PROGRESS result:-1)
03-29 14:00:51.041 26480 28560 V UploadsStorageManager: Updating /storage/emulated/0/DCIM/**/*.jpg with status=UPLOAD_IN_PROGRESS
03-29 14:00:51.072 26480 28560 D UploadsStorageManager: updateUpload returns with: 1 for file: /storage/emulated/0/DCIM/**/*.jpg
03-29 14:00:51.072 26480 28560 D UploadsStorageManager: notifyObserversNow
03-29 14:00:51.138 26480 26480 D UploadListAdapter: loadUploadItemsFromDb
03-29 14:00:51.150 26480 28560 V UploadsStorageManager: Updating /storage/emulated/0/DCIM/**/*.jpg with status=UPLOAD_IN_PROGRESS
03-29 14:00:51.160   439  1287 I BufferQueueProducer: [com.nextcloud.client/com.owncloud.android.ui.activity.UploadListActivity#0](this:0x783bd6a800,id:1345,api:1,p:26480,c:439) queueBuffer: fps=0.18 dur=5411.75 max=5411.75 min=5411.75
03-29 14:00:51.168 26480 28560 D UploadsStorageManager: updateUpload returns with: 1 for file: /storage/emulated/0/DCIM/**/*.jpg
03-29 14:00:51.168 26480 28560 D UploadsStorageManager: notifyObserversNow
03-29 14:00:51.216 26480 28560 D UploadFileOperation: normal upload
03-29 14:00:51.229 26480 28560 D AdvancedSslSocketFactory: Creating SSL Socket with remote ***.feste-ip.net:443, local null:0, params: org.apache.commons.httpclient.params.HttpConnectionParams@b41c1f6
03-29 14:00:51.230 26480 28560 D AdvancedSslSocketFactory:  ... with connection timeout 0 and socket timeout 0
03-29 14:00:51.234 26480 28560 I ServerNameIndicator: SNI done, hostname: ***.feste-ip.net
03-29 14:00:51.333 26480 26480 D UploadListAdapter: loadUploadItemsFromDb
03-29 14:00:51.340   435   478 I libPerfService: perfUserGetCapability - value:26480
03-29 14:00:51.344 26480 28560 V NativeCrypto: SSL handshake aborted: ssl=0x7a9a8b5048: I/O error during system call, Connection reset by peer
03-29 14:00:51.355 26480 28560 E com.nextcloud.client.network.ConnectivityServiceImpl: Error checking internet connection
03-29 14:00:51.365 26480 28560 E UploadFileOperation: Upload of /storage/emulated/0/DCIM/**/*.jpg to /**/*.jpg: No network connection
03-29 14:00:51.366 26480 28560 D UploadsStorageManager: updateDatabaseUploadResult uploadResult: RemoteOperationResult(mSuccess=false, mHttpCode=-1, mHttpPhrase=null, mException=null, mCode=NO_NETWORK_CONNECTION, getLogMessage=No network connection) upload: com.owncloud.android.operations.UploadFileOperation@1acea94
03-29 14:00:51.374 26480 28560 V UploadsStorageManager: Updating /storage/emulated/0/DCIM/**/*.jpg with status:UPLOAD_FAILED and result:NETWORK_CONNECTION (old:/storage/emulated/0/DCIM/**/*.jpg status:UPLOAD_IN_PROGRESS result:-1)
03-29 14:00:51.374 26480 28560 V UploadsStorageManager: Updating /storage/emulated/0/DCIM/**/*.jpg with status=UPLOAD_FAILED
03-29 14:00:51.386 26480 28560 D UploadsStorageManager: updateUpload returns with: 1 for file: /storage/emulated/0/DCIM/**/*.jpg
03-29 14:00:51.386 26480 28560 D UploadsStorageManager: notifyObserversNow
03-29 14:00:51.387 26480 28560 D FileUploader: NotifyUploadResult with resultCode: NO_NETWORK_CONNECTION
03-29 14:00:51.404 26480 28560 D FileUploader: Stopping command after id 73
```
</details>

**Comment by tobiasKaminsky** [comment]
Can you try to setup another account on the same device for testing?

**Comment by buehlefs** [comment]
Yes, how should I send you the credentials?

**Comment by buehlefs** [comment]
I've found my problem.
My fritzbox handed out a new ipv6 to my nextcloud server without updating the address in its firewall. Therefore no connection from outside would be able to connect to the nextcloud. (I did not find that out sooner because of the stay home situation right now..., though it may be a good idea to include the ip address that failed along with a not for a firewall that may have blocked the connection into the logs)
The app should have worked correctly using the ipv6 address as it was in the same network as the nextcloud server. But it seams to have used the ipv4 address from the port forwarding service I use which means the connection was from outside the network.
So this is actually related to #3193 and the issue can be closed.
